Android-NDK/jniapi.cpp

```diff
@@ -55,6 +69,28 @@ jobject wrapNode(JNIEnv* jenv, NodePointer node) {
     return wrapper;
 }
 
+//helper function for wrappers returning a traceroute probe result
+jobject wrapProbe(JNIEnv* jenv,
+                  std::string probeAddress,
+                  bool success,
+                  double elapsedMs) {
+
+    //strings that need to be freed after
+    jstring from = jenv->NewStringUTF(probeAddress.c_str());
+    jclass probeWrapperClass = jenv->FindClass("com/peer1/internetmap/ProbeWrapper");
+
+    jmethodID constructor = jenv->GetMethodID(probeWrapperClass, "<init>", "(ZLjava/lang/String;D)V");
+    //note: if you change this code, triple-check that the argument order matches ProbeWrapper.
+    probeWrapper = jenv->NewObject(probeWrapperClass, constructor, success, from, elapsedMs);
+
+    //free up the strings
+    jenv->DeleteLocalRef(from);
+    //oh, we need to free this too
+    jenv->DeleteLocalRef(probeWrapperClass);
```

Comment:
Based on the local / global ref docs I found when validating the probeWrapper behaviour (https://docs.oracle.com/javase/6/docs/technotes/guides/jni/spec/design.html#wp1242), I don't think these calls are required. These should be freed automatically when this function returns.